sdk/apimanagement/Azure.ResourceManager.ApiManagement/src/Customized/Models/RequestReportRecordContract.cs

```diff
@@ -0,0 +1,21 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+#nullable disable
+
+using System;
+using System.ComponentModel;
+using System.Net;
+using Azure.Core;
+
+namespace Azure.ResourceManager.ApiManagement.Models
+{
+    /// <summary> Request Report data. </summary>
+    public partial class RequestReportRecordContract
+    {
+        /// <summary> The HTTP status code received by the gateway as a result of forwarding this request to the backend. </summary>
+        [EditorBrowsable(EditorBrowsableState.Never)]
+        [Obsolete("This property has been replaced by BackendResponseCodeInteger", false)]
+        public string BackendResponseCode => BackendResponseCodeInteger.ToString();
```

Comment:
I kind of prefer we do not rename the property, instead, we introduce the `[CodeGenMemberSerializationHook]` attribute to this property to make sure it could properly serialize and deserialize.
For instance, in this case, previously we are deserializing this property using `property.Value.GetString`, using this attribute, we could change it to use:
`property.Value.GetInt32().ToString()`
this could leave our public API unchanged, but fix the JsonException

Reply:
You could find some document here: https://github.com/Azure/autorest.csharp/blob/feature/v3/readme.md#change-the-implementation-of-serializationdeserialization-method-of-one-particular-property

Reply:
Resolved.